src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Debug/StartupProjectRegistrar.cs

```diff
@@ -0,0 +1,173 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.ComponentModel.Composition;
+using System.Threading.Tasks.Dataflow;
+using Microsoft.VisualStudio.ProjectSystem.Debug;
+using Microsoft.VisualStudio.ProjectSystem.Utilities.DataFlowExtensions;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
+using Tasks = System.Threading.Tasks;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.Debug
+{
+    /// <summary>
+    /// <see cref="StartupProjectRegistrar"/> is responsible for adding or removing a project from the Startup list
+    /// depending on whether the active configuration of the a project is debuggable or not.
+    /// </summary>
+    internal class StartupProjectRegistrar :
```

Comment:
Please implement OnceInitializedOnceDisposed instead of handling "have I been disposed" logic yourself